src/components/Feed/Feed.tsx

```diff
@@ -0,0 +1,150 @@
+export const WORLDWIDE = Symbol('WORLDWIDE');
+export type FeedMode = typeof FOLLOWING | typeof WORLDWIDE;
+
+type ControlProps = {
+  setFeedMode: (mode: FeedMode) => void;
+  initialFeedMode: FeedMode;
+};
+
+function FeedNavbarControl({ setFeedMode, initialFeedMode }: ControlProps) {
+  // Internally mirror the feed mode state so that we can avoid adding it to the dep array, and avoid re-adding these controls to the navbar when the user switches from Following to Worldwide feed
+  const [feedModeCopy, setFeedModeCopy] = useState<FeedMode>(initialFeedMode);
+
+  const handleFollowingModeClick = useCallback(() => {
+    setFeedMode(FOLLOWING);
+    setFeedModeCopy(FOLLOWING);
+  }, [setFeedMode]);
+
+  const handleWorldwideModeClick = useCallback(() => {
+    setFeedMode(WORLDWIDE);
+    setFeedModeCopy(WORLDWIDE);
+  }, [setFeedMode]);
+
+  return (
+    <StyledFeedNavbarControl>
+      <StyledTextWrapperLeft>
+        <StyledNavControlText
+          onClick={handleFollowingModeClick}
+          focused={feedModeCopy === FOLLOWING}
+        >
+          Following
+        </StyledNavControlText>
+      </StyledTextWrapperLeft>
+      <Spacer width={10} />
+      <NavbarGLink />
+      <Spacer width={10} />
+      <StyledTextWrapper>
+        <StyledNavControlText
+          onClick={handleWorldwideModeClick}
+          focused={feedModeCopy === WORLDWIDE}
+        >
+          Worldwide
+        </StyledNavControlText>
+      </StyledTextWrapper>
+    </StyledFeedNavbarControl>
+  );
+}
+
+const StyledNavControlText = styled(ActionText)<{ focused: boolean }>`
+  animation: ${fadeIn} ${transitions.cubic} 700ms 1 forwards;
+  color: ${({ focused }) => (focused ? colors.offBlack : colors.metal)};
+
+  opacity: 0;
+`;
+
+const StyledTextWrapper = styled.div`
+  display: flex;
+  flex: 1;
+`;
+
+const StyledTextWrapperLeft = styled(StyledTextWrapper)`
+  justify-content: flex-end;
+`;
+
+const StyledFeedNavbarControl = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+`;
+
+export default function Feed() {
+  const query = useLazyLoadQuery<FeedViewerQuery>(
+    graphql`
+      query FeedViewerQuery {
+        viewer {
+          ... on Viewer {
+            user {
+              dbid
+            }
+            # ...FeedNavbarViewerFragment
+          }
+        }
+      }
+    `,
+    {}
+  );
+  const { viewer } = query;
+  const viewerUserId = (viewer?.user?.dbid ?? '') as string;
+  const defaultFeedMode = viewerUserId ? FOLLOWING : WORLDWIDE;
+
+  const [feedMode, setFeedMode] = useState<FeedMode>(defaultFeedMode);
+  const { setCustomNavCenterContent } = useGlobalLayoutActions();
+
+  // This effect ensures the Feed controls on the navbar are removed when the Feed unmounts, so that it is not visible when navigating to a different page.
+  useEffect(() => {
+    return () => {
+      setTimeout(() => {
+        setCustomNavCenterContent(null);
+      }, 200);
```

Comment:
we should be able to use `FADE_TRANSITION_TIME_MS` which is 300ms, used by the navbar. unless the interaction is weird

Reply:
done